src/firefly/js/visualize/ui/extraction/ExtractionTable.jsx

```diff
@@ -84,46 +91,186 @@ export function keepZAxisExtraction(pt, pv, plot, filename, refHDUNum, extractio
     save ? doDispatchTableSaving(dataTableReq, doOverlay) : doDispatchTable(dataTableReq, doOverlay);
     idCnt++;
     titleCnt++;
-    return tbl_id;
+    return Promise.resolve([tbl_id]);
 }
 
-export function keepLineExtraction(pt, pt2, pv, plot, filename, refHDUNum, plane, extractionSizeX, extractionSizeY, combineOp, save = false, doOverlay = true) {
+export function keepDataExtraction({pv, baseImPtAry, save, doOverlay, axis='both', pointSize, combineOp, isLine=false}) {
+    const plot= primePlot(pv);
+    const filename= plot?.plotState?.getWorkingFitsFileStr();
     if (!pv || !plot || !filename) {
         showInfoPopup('Plot no longer exist. Cannot extract.');
         return;
     }
+    const {pvAry,filteredPvAry,canDoMultiImage}= getExtractableImageList(pv);
+    const extractionSizeX= axis==='x' ? 1 : pointSize;
+    const extractionSizeY= axis==='y' ? 1: pointSize;
+    const params= {baseImPtAry, pv, extractionSizeX, extractionSizeY, combineOp, save, doOverlay, isLine};
+    if (!canDoMultiImage) {
+        return Promise.resolve([makeDataExtractionTable(params)]);
+    }
+
+    if (filteredPvAry.length < pvAry.length) {
+        let msg= 'Cannot extract from all images';
+        if (pvAry.some( (testPv) => !hasWCSProjection(testPv))) {
+            msg+= ' Some image do not have a WCS Projection.';
+        }
+        if (pvAry.some( (testPv) => getMatchingHDUCount(testPv)>MAX_HDU)) {
+            msg+= ` Some images have more than ${MAX_HDU} HDUs.`;
+        }
+        msg+= ' These image can be extracted in single image mode';
+        setTimeout( () => showInfoPopup(msg), 1000);
+    }
+
+
+    const AllImagesQuestion= () => (
+        <Stack spacing={2} sx={{width:'30rem'}}>
+            <Typography>
+                You can either extract from the currently selected single image, or all loaded images that have a WCS
+                and a limited amount of HDUs (with WCS information).
+            </Typography>
+            <Typography color='warning'>
+                Do you want to extract from all images?
+            </Typography>
+        </Stack>
+    );
+    const answers= {allSame: 'Yes / into one table', allDiff: 'Yes / into multiple tables', single: 'No'};
+
```

Comment:
I think this should be: 

`Yes - into one table`, `Yes - into multiple tables` **_OR_**
`Yes (into one table)`, `Yes (into multiple tables)` 

as `/` may suggest a choice between the two, but I'll defer this to @lrebull

Reply:
I mostly just used it like that to make the `yes` stand out more. You have a good point. maybe @lrebull need to weigh in again on this.